# Analysis of Financial Markets based on President Trump's Tweets

## Data Preprocessing

### Importing Data from Kaggle

In [1]:
!pip install -q kaggle

In [ ]:
# Upload kaggle.json file containing your API key
from google.colab import files
files.upload()

In [ ]:
!mkdir ~/.kaggle 

In [4]:
!cp kaggle.json ~/.kaggle/

In [5]:
! chmod 600 ~/.kaggle/kaggle.json

In [ ]:
# Check to see if the API is working correctly 
# ! kaggle datasets list

In [ ]:
! kaggle datasets download -d austinreese/trump-tweets

### Exploritory Analysis

In [33]:
import pandas as pd
import numpy as np
from zipfile import ZipFile

In [34]:
data = ZipFile("/content/trump-tweets.zip")
trump_tweets = pd.read_csv(data.open('trumptweets.csv'))
trump_tweets.head()

id                                               link  ... hashtags geo
0  1698308935  https://twitter.com/realDonaldTrump/status/169...  ...      NaN NaN
1  1701461182  https://twitter.com/realDonaldTrump/status/170...  ...      NaN NaN
2  1737479987  https://twitter.com/realDonaldTrump/status/173...  ...      NaN NaN
3  1741160716  https://twitter.com/realDonaldTrump/status/174...  ...      NaN NaN
4  1773561338  https://twitter.com/realDonaldTrump/status/177...  ...      NaN NaN

[5 rows x 9 columns]

In [35]:
trump_tweets.shape

(41122, 9)

The dataframe shows that there are 41,122 rows and 9 columns. Three of these columns mention, hashtags, and geo are currenly showing NaN for the first five rows. 

In [36]:
trump_tweets.describe()

id       retweets      favorites  geo
count  4.112200e+04   41122.000000   41122.000000  0.0
mean   6.088909e+17    5455.590657   22356.899105  NaN
std    3.027946e+17   10130.076661   41501.859711  NaN
min    1.698309e+09       0.000000       0.000000  NaN
25%    3.549428e+17      25.000000      28.000000  NaN
50%    5.609149e+17     291.000000     247.000000  NaN
75%    7.941218e+17    8778.000000   32970.750000  NaN
max    1.219077e+18  309892.000000  857678.000000  NaN

In [37]:
#Remove NaN columns
trump_tweets = trump_tweets.drop(labels=['mentions', 'hashtags', 'geo'], axis='columns')
trump_tweets

id  ... favorites
0               1698308935  ...       868
1               1701461182  ...       273
2               1737479987  ...        18
3               1741160716  ...        24
4               1773561338  ...      1965
...                    ...  ...       ...
41117  1218962544372670467  ...    213817
41118  1219004689716412416  ...    149571
41119  1219053709428248576  ...     66944
41120  1219066007731310593  ...     81921
41121  1219076533354037249  ...     34902

[41122 rows x 6 columns]

In [38]:
trump_tweets.dtypes

id            int64
link         object
content      object
date         object
retweets      int64
favorites     int64
dtype: object

In [39]:
# The date column is an object when it should be datetime
trump_tweets['date'] = pd.to_datetime(trump_tweets['date'], format='%Y%m%d %H:%M:%S')

In [43]:
# Split the date column into 2 seperate columns
trump_tweets['Time'],trump_tweets['Date']= trump_tweets['date'].apply(lambda x:x.time()), trump_tweets['date'].apply(lambda x:x.date())
trump_tweets = trump_tweets.drop(labels='date',axis='columns')

In [44]:
trump_tweets.head()

id  ...        Date
0  1698308935  ...  2009-05-04
1  1701461182  ...  2009-05-05
2  1737479987  ...  2009-05-08
3  1741160716  ...  2009-05-08
4  1773561338  ...  2009-05-12

[5 rows x 7 columns]